source/opt/ir_context.cpp

```diff
@@ -106,6 +106,10 @@ Instruction* IRContext::KillInst(ir::Instruction* inst) {
     type_mgr_->RemoveId(inst->result_id());
   }
 
+  if (constant_mgr_ && inst->IsConstant()) {
```

Comment:
The constant manager can generate a constant from a OpSpecConstantComposite (at least in some cases). This check will not identify those. Perhaps the cleanest way would be to add a method to constant manager to indicate supported constant opcodes? It would return true for IsConstant() and OpSpecConstantComposite.

Reply:
Restricting just to OpSpecConstantComposite seems slightly odd - would it make sense to use ir::IsConstantInst(inst->opcode()) instead?

Reply:
The oddness is why I propose the method test method belong to ConstantManager. Then it can completely own the implementation weirdness; however, RemoveId looks safe for all ids so I'm also ok with your suggestion.